src/Feature/Theme/GlobalColors.re

```diff
@@ -0,0 +1,216 @@
+  let visualModeForeground = "oni.visualModeForeground";
+  let insertModeForeground = "oni.insertModeForeground";
+  let replaceModeForeground = "oni.replaceModeForeground";
+  let normalModeForeground = "oni.normalModeForeground";
+  let operatorModeForeground = "oni.operatorModeForeground";
+  let commandlineModeForeground = "oni.commandlineModeForeground";
+
+  let backgroundFor = (mode: Vim.Mode.t) =>
+    switch (mode) {
+    | Visual => visualModeBackground
+    | CommandLine => commandlineModeBackground
+    | Operator => operatorModeBackground
+    | Insert => insertModeBackground
+    | Replace => replaceModeBackground
+    | Normal => normalModeBackground
+    };
+
+  let foregroundFor = (mode: Vim.Mode.t) =>
+    switch (mode) {
+    | Visual => visualModeForeground
+    | CommandLine => commandlineModeForeground
+    | Operator => operatorModeForeground
+    | Insert => insertModeForeground
+    | Replace => replaceModeForeground
+    | Normal => normalModeForeground
+    };
+
+  let defaults =
+    ColorTheme.Defaults.[
+      (visualModeBackground, hex("#56b6c2") |> uniform),
+      (insertModeBackground, hex("#98c379") |> uniform),
+      (replaceModeBackground, hex("#d19a66") |> uniform),
+      (normalModeBackground, hex("#61afef") |> uniform),
+      (operatorModeBackground, hex("#d19a66") |> uniform),
+      (commandlineModeBackground, hex("#61afef") |> uniform),
+      (visualModeForeground, hex("#282c34") |> uniform),
+      (insertModeForeground, hex("#282c34") |> uniform),
+      (replaceModeForeground, hex("#282c34") |> uniform),
+      (normalModeForeground, hex("#282c34") |> uniform),
+      (operatorModeForeground, hex("#282c34") |> uniform),
+      (commandlineModeForeground, hex("#282c34") |> uniform),
+    ];
+};
+
+module SideBar = {
+  let background = "sidebar.background";
+  let foreground = "sidebar.foreground";
+
+  let defaults =
+    ColorTheme.Defaults.[
+      (background, hex("#21252b") |> uniform),
+      (foreground, hex("#ECEFF4") |> uniform),
+    ];
+};
+
+module Tab = {
+  let activeBackground = "tab.activeBackground";
+  let activeForeground = "tab.activeForeground";
+  let inactiveBackground = "tab.inactiveBackground";
+  let inactiveForeground = "tab.inactiveForeground";
+
+  let defaults =
+    ColorTheme.Defaults.[
+      (
+        activeBackground,
+        {
+          dark: ref(Editor.background),
+          light: ref(Editor.background),
+          hc: ref(Editor.background),
+        },
+      ),
+      (
+        activeForeground,
+        {dark: hex("#fff"), light: hex("#333"), hc: hex("#fff")},
+      ),
+      (
+        inactiveBackground,
+        {dark: hex("#2D2D2D"), light: hex("#ECECEC"), hc: unspecified},
+      ),
+      (
+        inactiveForeground,
+        {
+          dark: transparent(0.4, ref(activeForeground)),
+          light: transparent(0.4, ref(activeForeground)),
+          hc: hex("#fff"),
+        },
+      ),
```

Comment:
This is one of the more advanced definitions.

Reply:
Cool! You've done a nice job expressing the different (unhandled) cases that come up in VSCode themes - like the references and derived theme colors.